### PR TITLE
Unit tests added for MaintenancePlugin

### PR DESCRIPTION
**Pull request by DeepDiver1975**
unit tests for #17226
@LukasReschke @MorrisJobke

### CODE REVIEW

File: lib/private/connector/sabre/maintenanceplugin.php [hunk 1]
```diff
@@ -39,6 +43,17 @@ class MaintenancePlugin extends ServerPlugin {
 	private $server;
 
 	/**
+	 * @param IConfig $config
+	 */
+	public function __construct(IConfig $config) {
```
Comment (nickvergessen) [inline lib/private/connector/sabre/maintenanceplugin.php]: https://github.com/owncloud/core/blob/master/apps/files/appinfo/remote.php#L53-53
https://github.com/owncloud/core/blob/master/apps/files_sharing/publicwebdav.php#L49-49

Comment (nickvergessen) [inline lib/private/connector/sabre/maintenanceplugin.php]: I see you have a check for that. But I'd say adjust the calls in core at least?

Comment (DeepDiver1975) [inline lib/private/connector/sabre/maintenanceplugin.php]: default value null is missing

**Comment by DeepDiver1975** [comment]
@nickvergessen please review once more - thanks for catching the missing changes

**Comment by MorrisJobke** [comment]
Tested again and it still works :+1:

**Comment by nickvergessen** [comment]
:+1:

**Comment by MorrisJobke** [comment]
@nickvergessen CI is still running ;)

**Comment by nickvergessen** [comment]
@MorrisJobke while this is true, the passing tests before the fix only confirm, that we don't have tests for the part that did not work before the second commit, so nothing should change.

**Comment by LukasReschke** [comment]
:bomb: :bomb: :bomb:

```
5:13 <PVince81> {"reqId":"ba4hRkmJdlHcA4FJWfQS","remoteAddr":"127.0.0.1","app":"PHP","message":"Cannot use Sabre\\DAV\\Server as Server because the name is already in use at \/srv\/www\/htdocs\/owncloud\/lib\/private\/connector\/sabre\/maintenanceplugin.php#30","level":3,"time":"2015-06-29T14:54:30+00:00","method":"PROPFIND","url":"\/owncloud\/remote.php\/webdav\/"}
```

Really folks… Come on.

**Comment by nickvergessen** [comment]
@LukasReschke wrong PR: https://github.com/owncloud/core/pull/17226/files#diff-d3eea75321ad0330101e9c5bf2d54b12R29

**Pull request merged**